com.ibm.wala.cast.python/source/com/ibm/wala/cast/python/ipa/callgraph/PythonSSAPropagationCallGraphBuilder.java

```diff
@@ -211,7 +218,19 @@ protected void processCallingConstraints(
           }
         } else {
           PointerKey rval = getPointerKeyForLocal(caller, call.getUse(i));
-          getSystem().newConstraint(lval, assignOperator, rval);
+
+          // If we are looking at the implicit parameter of a callable.
+          if (call.getCallSite().isDispatch() && i == 0 && isCallable(rval)) {
+            // Ensure that lval's variable refers to the callable method instead of callable object.
+            IClass callable = getCallable(target);
+            IntSet instanceKeysForCallable = this.getSystem().getInstanceKeysForClass(callable);
+
+            for (IntIterator it = instanceKeysForCallable.intIterator(); it.hasNext(); ) {
+              int instanceKeyIndex = it.next();
+              InstanceKey instanceKey = this.getSystem().getInstanceKey(instanceKeyIndex);
+              this.getSystem().newConstraint(lval, instanceKey);
+            }
+          } else getSystem().newConstraint(lval, assignOperator, rval);
```

Comment:
(Nit) Can we enclose the `else` case in braces for consistency?

Reply:
Nope. Spotless this according to the Google format.

Reply:
I'll try it just to make sure.